pytensor/link/mlx/dispatch/core.py

```diff
@@ -0,0 +1,253 @@
+"""
+pytensor/link/mlx/dispatch/basic.py
+-----------------------------------
+
+First-cut MLX translations for the most common tensor Ops.
+
+The structure intentionally follows pytensor's JAX dispatcher so that
+once these kernels stabilise they can be optimised further (e.g. fusing
+element-wise graphs, adding in-place updates, RNG thinning, etc.).
+"""
+
+from __future__ import annotations
+
+import warnings
+
+import mlx.core as mx  # MLX
+import numpy as np
+
+from pytensor.link.mlx.dispatch.basic import mlx_funcify  # MLX
+from pytensor.tensor import get_vector_length
+from pytensor.tensor.basic import (
+    Alloc,
+    AllocEmpty,
+    ExtractDiag,
+    Eye,
+    Join,
+    MakeVector,
+    ScalarFromTensor,
+    Split,
+    TensorFromScalar,
+    Tri,
+    get_scalar_constant_value,
+)
+from pytensor.tensor.exceptions import NotScalarConstantError
+
+
+# ------------------------------------------------------------------
+# Join
+# ------------------------------------------------------------------
+@mlx_funcify.register(Join)  # MLX
+def mlx_funcify_Join(op, **kwargs):
+    def join(axis, *tensors):
+        view = op.view
+        if (view != -1) and all(
+            tensors[i].shape[axis] == 0  # MLX
+            for i in list(range(view)) + list(range(view + 1, len(tensors)))
+        ):
+            return tensors[view]
+
+        return mx.concatenate(tensors, axis=axis)  # MLX
+
+    return join
+
+
+# ------------------------------------------------------------------
+# Split
+# ------------------------------------------------------------------
+@mlx_funcify.register(Split)  # MLX
+def mlx_funcify_Split(op: Split, node, **kwargs):
+    _, axis_sym, splits_sym = node.inputs
+
+    try:
+        constant_axis = get_scalar_constant_value(axis_sym)
+    except NotScalarConstantError:
+        constant_axis = None
+        warnings.warn(
+            "Split node does not have a constant axis. MLX implementation may fail."
+        )
+
+    try:
+        constant_splits = np.array(
+            [
+                get_scalar_constant_value(splits_sym[i])
+                for i in range(get_vector_length(splits_sym))
+            ]
+        )
+    except (ValueError, NotScalarConstantError):
+        constant_splits = None
+        warnings.warn(
+            "Split node does not have constant split positions. MLX implementation may fail."
```

Comment:
Sounds like JAX stuff. Anything that seems like a backend restriction should not be assumed to be needed in MLX by default